pkg/operator/ceph/object/bucket/rgw-handlers.go

```diff
@@ -47,15 +50,25 @@ func (p *Provisioner) createCephUser(username string) (accKey string, secKey str
 	return u.Keys[0].AccessKey, u.Keys[0].SecretKey, nil
 }
 
+func random(fsid string) string {
+	randString := ""
+	for range fsid {
+		rand, _ := rand.Int(rand.Reader, big.NewInt(int64(len(fsid))))
+		randString += string(fsid[rand.Int64()])
+	}
+	return string(randString)
+}
+
 func (p *Provisioner) genUserName(obcName, obcNamespace string) (string, error) {
```

Comment:
This implements the spec for lib-bucket-provisioner's `GenerateUserID` which says

```
	// GenerateUserID should deterministically generate a user ID for an OBC. This ID is used as an
	// idempotency key in order to ensure repeat calls to Provision or Grant are consistent.
	// Lib-bucket-provisioner may pass a non-nil ObjectBucket with the function as well. This will
	// be nil if the ObjectBucket does not yet exist, but it will be non-nil for existing buckets.
	// This may help provisioners recover the idempotency key from a pre-existing bucket.
	GenerateUserID(obc *v1alpha1.ObjectBucketClaim, ob *v1alpha1.ObjectBucket) (string, error)
```

Reply:
So I should append the UID?

Reply:
1. Is there any issue with a really long user ID? If we append the UID, it will be quite long. For example, some K8s resource names can only be a certain length.
2. If we append the UID, will that cause issues during upgrade with existing users? We will effectively be changing the ID on them. But if #11665 was able to update it and we didn't see issues, perhaps it is fine.

Reply:
I think I asked @alimaredia 

> 1. Is there any issue with a really long user ID? If we append the UID, it will be quite long. For example, some K8s resource names can only be a certain length.

I think I asked @alimaredia about this and learned that the usernames can be up to 255 chars IIRC.

> 2. If we append the UID, will that cause issues during upgrade with existing users? We will effectively be changing the ID on them. But if [object: make OBC genUserID unique across clusters #11665](https://github.com/rook/rook/pull/11665) was able to update it and we didn't see issues, perhaps it is fine.

I don't believe this is an issue. l-b-p should use the ObjectBucket's bucket name field once it exists. By that behavior, existing buckets should continue to work. I think I tested that when I made the code changes, but it's still a great thing to test again. More validation here won't hurt.

GenerateUserID will is used while l-b-p is attempting to provision a bucket. The ID should be deterministic so that any failures in Ceph or RGW (or other distributed components) don't result in hundreds/thousands of unused users being created due to the attempted-but-failed provisioning of a single ObjectBucketClaim. I'll keep looking for the Rook issue where this was reported as a bug, but I also saw that behavior/bug in some of my testing.

Reply:
Another user has a unique idea of using the OBC's UID as part of the user ID. This could be a good missing piece of the puzzle. https://github.com/rook/rook/pull/12884

@haslersn pointed out that these UID are considered universally unique by k8s spec, so we shouldn't need any more uniqueness. And it is also deterministic for any given OBC. I think Sebastian's PR is probably the easier solution here.